packages/nx/src/lock-file/npm-parser.ts

```diff
@@ -580,10 +580,20 @@ function elevateNestedPaths(
       `${segs.join('/node_modules/')}/node_modules/${packageName}`;
 
     // check if grandparent has the same package
-    while (
-      segments.length > 1 &&
-      !result.has(getNewPath(segments.slice(0, -1)))
-    ) {
+    const shouldElevate = (segs: string[]) => {
+      const newPath = getNewPath(segs.slice(0, -1));
+      if (result.has(newPath)) {
+        const match = result.get(newPath);
+        const source = remappedPackages.get(path);
+        return (
+          match.valueV1?.version === source.valueV1?.version &&
```

Comment:
If a parent already exists, check if it's the same version. In which case, it's ok to pop the segment and shorten the path.